modules/client/src/client-common/context.ts

```
@@ -1,116 +1,151 @@
+      this.state.network = Context.resolveNetwork(
+        contextParams.network,
+        contextParams.ensRegistryAddress,
+      );
+      // once the network is resolved set default values
+      this.setDefault();
     }
     if (contextParams.signer) {
       this.state.signer = contextParams.signer;
     }
-    if (contextParams.web3Providers) {
+    if (
+      contextParams.web3Providers ||
+      (Array.isArray(contextParams.web3Providers) &&
+        contextParams.web3Providers.length)
+    ) {
       this.state.web3Providers = Context.resolveWeb3Providers(
         contextParams.web3Providers,
-        this.state.network
+        this.state.network,
       );
+      this.manualState.web3 = true;
+    }
+    if (contextParams.graphqlNodes?.length) {
+      this.state.graphql = Context.resolveGraphql(contextParams.graphqlNodes);
+      this.manualState.graphql = true;
+    }
+    if (contextParams.ipfsNodes?.length) {
+      this.state.ipfs = Context.resolveIpfs(contextParams.ipfsNodes);
+      this.manualState.ipfs = true;
+    }
+    if (contextParams.daoFactoryAddress) {
+      this.state.daoFactoryAddress = contextParams.daoFactoryAddress;
+      this.manualState.daoFactory = true;
+    }
+    if (contextParams.ensRegistryAddress) {
+      this.state.ensRegistryAddress = contextParams.ensRegistryAddress;
+      this.manualState.ensRegistry = true;
     }
     if (contextParams.gasFeeEstimationFactor) {
       this.state.gasFeeEstimationFactor = Context.resolveGasFeeEstimationFactor(
-        contextParams.gasFeeEstimationFactor
+        contextParams.gasFeeEstimationFactor,
       );
+      this.manualState.gasFeeEstimationFactor = true;
     }
+  }
 
-    if (contextParams.ipfsNodes?.length) {
-      this.state.ipfs = Context.resolveIpfs(contextParams.ipfsNodes);
+  private setDefault() {
+    const networkName = this.state.network.name as SupportedNetworks;
+    if (
+      WEB3_NODES[networkName] && WEB3_NODES[networkName].length &&
```

If the length was zero I would rather throw an internal error

---

Sure :)